[MCC-106990] Benchmark Deserialization

kenyamat opened the pull request:
@mdsol/hypermedia-team - I created a benchmarking tool.
Please review this.

kenyamat committed:
Deleted AllowPartiallyTrustedCallers attribute.

kenyamat committed:
Added Crichton.Representors.Benchmark.

edandersen commented:
Run.bat should:
- be in the root of the repository, maybe called "benchmark.bat"
- actually compile the project with MSBuild in Release mode before running the exe

edandersen commented on tools/Crichton.Representors.Benchmark/run.bat:
```diff
@@ -0,0 +1,2 @@
+.\bin\Debug\Crichton.Representors.Benchmark.exe deserializer --iterations 5 --filepath "C:\GitProjects\crichton-dotnet\tests\Crichton.Representors.Tests\Integration\TestData\Hal\AllLinkObjectProperties.json"
```
Should be a relative path here to the JSON instead of absolute

edandersen commented:
Also lets put the benchmark tool project in a Solution Folder called "tools" in VS.

kenyamat committed:
Modified feedbacks from Ed.
* be in the root of the repository, maybe called "benchmark.bat"
* actually compile the project with MSBuild in Release mode before running the exe
* Should be a relative path here to the JSON instead of absolute
* Also lets put the benchmark tool project in a Solution Folder called "tools" in VS.
Added README.md for Benchmark.

kenyamat committed:
Modified README.md.

kenyamat commented:
@edandersen Thank you for your feedback.
I modified the issues and also added README.md.

edandersen commented:
@kenyamat Thanks for the updates. I checked out the branch and ran `benchmark.bat` and it successfully built the libraries and ran the benchmark.
- Please remove the "press any key to continue" Console.ReadKey as this will block if we run this in CI
- Please move the README.md to the /docs folder and add a link to it on the main project Readme,md

...then it's good to merge!

kenyamat committed:
Moved and renamed README.md into representor_benchmark_tool.md.
Remove pause statement in benchmark.bat.

kenyamat commented:
Hi @edandersen , Thank you for your review.
The code was fixed. Could you please review this again?
- Moved and renamed README.md into representor_benchmark_tool.md.
- Removed pause statement in benchmark.bat.

edandersen commented:
Awesome! Merging.
